include/bluetooth/mesh/scene_srv.rst

```diff
@@ -0,0 +1,108 @@
+
+* Scene Server
+* Scene Setup Server
+
+The two model instances share the states of the Scene Server, but accept different messages.
+This allows for a fine-grained control of the access rights for the Scene Server states, as the two model instances can be bound to different application keys.
+
+* Scene Server provides functionality for listing and recalling stored scenes.
+* Scene Setup Server provides functionality for storing and deleting scenes.
+
+Operation
+=========
+
+The Scene Server stores all scene data persistently using the :ref:`zephyr:settings_api` subsystem.
+Every scene is stored as a serialized concatenation of each registered model's state, and only exists in RAM during storing and loading.
+
+It's up to the individual model implementation to correctly serialize and deserialize its state from scene data when prompted.
+
+Models with Scene data
+**********************
+
+The following models will automatically register as Scene entries:
+
+* :ref:`bt_mesh_onoff_srv_readme`
+* :ref:`bt_mesh_lvl_srv_readme`
+* :ref:`bt_mesh_light_ctl_srv_readme`
+* :ref:`bt_mesh_light_ctrl_srv_readme`
+
+All models that extend these models will automatically become part of the scenes as well.
+
+Adding Scene data for a model
+*****************************
+
+Every model that needs to store its state in scenes must register a :cpp:type:`bt_mesh_scene_entry` structure containing callbacks for storing and recalling the scene data.
+The Scene entry must be registered with valid a :cpp:type:`bt_mesh_scene_entry_type` structure, which defines the Scene entry store and recall behavior.
+
+For instance, the Generic OnOff Server model defines its Scene entry by adding a `bt_mesh_scene_entry` structure to every :cpp:type:`bt_mesh_onoff_srv`, and registering it with a common type:
+
+.. literalinclude:: ../../../subsys/bluetooth/mesh/gen_onoff_srv.c
+   :language: c
+   :start-after: include_startingpoint_scene_srv_rst_1
+   :end-before: include_endpoint_scene_srv_rst_1
+
+The Generic OnOff Server model only needs to store its OnOff state in the scene, so it always stores and recalls a single byte.
+The ``restore`` call contains a transition time pointer, which should be taken into account when implementing the model behavior.
+
+If a Scene store operation is requested in the middle of a transition between two states, the model should commit the target state to the scene data.
+
+Invalidating Scene data
+***********************
+
+A Scene is only valid as long as all its state is unchanged.
+As soon as one of the models in a scene detect a change to the Scene data, either because of an incoming message or because of an internal change, the node is no longer in an active Scene.
+When a model detects a change to its own data, it shall call :cpp:func:`bt_mesh_scene_invalidate`, passing its own scene entry.
+This resets the Scene Server's current scene to :c:macro:`BT_MESH_SCENE_NONE`.
+The Scene has to be stored again to include the change.
```

Comment:
```suggestion
Invalidating scene data
***********************

A scene is valid as long as its state is unchanged.
As soon as one of the models in a scene detect a change to the scene data, either because of an incoming message or because of an internal change, the node is no longer in an active scene.
When a model detects a change to its own data, it shall call :cpp:func:`bt_mesh_scene_invalidate`, passing its own scene entry.
This resets the Scene Server's current scene to :c:macro:`BT_MESH_SCENE_NONE`.
The scene has to be stored again to include the change.
```